### ISSUE TITLE
nonstrict-import applies on packages in node_modules

### ISSUE THREAD

**Issue by VaclavSir** [facebook/flow#6628]
I'm not sure if it's a bug or a feature.

When I enable Flow Strict with `nonstrict-import` lint, then I can't import from libraries that export Flow files, but they are not strict.

For example the Facebook's IDX:

```
Dependencies of a @flow strict module must also be @flow strict! (nonstrict-import)

     1│ // @flow strict
     2│
     3│ import idx from 'idx';
     4│
```

But I can still import any module that has a flow-typed library definition. Isn't that strange? Libraries that are not even shipped with types are considered strict, libraries with types are non-strict.

**Comment by VaclavSir** [comment]
I realised IDX wasn't the best example, because its last version is strict...

But anyway, I think this behaviour is problematic, mainly for two reasons:

- Different authors use different set of lints and Flow doesn't support nested configs
- You can't expect that all libraries will adopt Flow Strict anytime soon, especially when they have many dependencies

**Comment by zeorin** [comment]
I just ran into this also. It'd be great if there were a sub-option available to exclude 3rd party modules (everything in the `module.system.node.resolve_dirname` folders) from the _strict_ requirement, like there are sub-options for the `sketchy-null` lint rule.

**Comment by fishythefish** [comment]
Yeah, perhaps `nonstrict-import` should imply `untyped-import`. (In the meantime, you may want to consider enabling that lint too.)

Would `@flow strict-local` be too weak for your use case?

**Comment by zeorin** [comment]
In my case, yes. Ideally I'd like to use `@flow strict`, but not be held up by 3rd party modules not implementing it yet (or ever).

**Comment by VaclavSir** [comment]
> Yeah, perhaps `nonstrict-import` should imply `untyped-import`. (In the meantime, you may want to consider enabling that lint too.)

I have untyped-import enabled too:

```
[strict]
nonstrict-import
unclear-type
untyped-import
untyped-type-import
unsafe-getters-setters
sketchy-null
```

> Would `@flow strict-local` be too weak for your use case?

Yes, it would. My goal is to have all files in the project `@flow strict` eventually.

For now, if only non-strict libraries are blocking me from using `@flow strict`, then I add a comment to the import with `// $FlowFixMe https://github.com/facebook/flow/issues/6628`, but I'm not very happy about it.

**Comment by josedarioxyz** [comment]
I'm wondering if there are any updates on this. Is there a way to ignore nonstrict-import warnings only from specified directories? Take an example from CRA. Say I want to keep my `src` folder internally consistent with `// @flow strict`, but I want non-strict imports from the node_modules directory outside of src to be allowed.

**Comment by zeorin** [comment]
I have worked around this for the time being by ignoring the modules in my `.flowconfig`, and then adding my own type definition in my project's `flow-typed` folder for it.

```
[ignore]
<PROJECT_ROOT>/node_modules/recompose
<PROJECT_ROOT>/node_modules/react-select
```

**Comment by pascalduez** [comment]
Same issue here.

One of our lib use a set of linting rules. It has a suppression comment in its code. 
`// flowlint-next-line unnecessary-optional-chain: off`
The consuming app doesn't have the linter activated, or a different set of rules.

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/.../Foo.js.flow:

Redundant argument. This argument doesn't change any lint settings.

17│
19│   // flowlint-next-line unnecessary-optional-chain: off
20│  some?.stuff
21│

```
